### ISSUE TITLE
Use online softmax for just softmax by itself

### ISSUE THREAD

**Issue by Chillee** [pytorch/pytorch#124145]
### 🚀 The feature, motivation and pitch

torch.compile currently uses the standard 3 pass algorithm for computing softmax. It would be neat for Inductor to codegen using the online softmax algorithm for sufficiently large softmaxes.

Could be done using similar infra to https://github.com/pytorch/pytorch/pull/102486
